#! /bin/sh -e

if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
elif [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch $pdir -f --no-backup-if-mismatch -p0 < $0;;
    -unpatch) patch $pdir -f --no-backup-if-mismatch -R -p0 < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

# DP: Fix for causing a segmentation fault when some special shell variables
# DP: are declared as array.

--- arrayfunc.c~	Mon Dec 31 03:32:24 2001
+++ arrayfunc.c	Wed Jan  9 17:18:28 2002
@@ -58,6 +58,10 @@
   FREE (value_cell (var));
   var->value = (char *)array;
 
+  /* these aren't valid anymore */
+  var->dynamic_value = (DYNAMIC_FUNC *)NULL;
+  var->assign_func = (DYNAMIC_FUNC *)NULL;
+
   INVALIDATE_EXPORTSTR (var);
 
   VSETATTR (var, att_array);
